### PR TITLE
feat(aws): introduced default chain auth

### PR DESCRIPTION
**Pull request by igpetrov**
## Description

feat(aws): introduced default chain auth

## CX

Screenshot 2023-10-23 at 21 18 32

Screenshot 2023-10-23 at 21 18 43

Screenshot 2023-10-23 at 21 18 54

## Testing done

1. Export environment variables:
- `AWS_ACCESS_KEY`
- `AWS_SECRET_KEY`
2. Test with and without credentials.
3. Both outbound and inbound.
4. Test previous version templates for backwards compatibility.

## Reason for draft
- I would like to get CX approved
- At the last moment, I see inbound SQS receives auth type `null` for default chain, need to debug why.

## Related issues

closes https://github.com/camunda/team-connectors/issues/479

### CODE REVIEW

**Comment by igpetrov** [comment]
Hi @YanaSegal @bastiankoerber @sbuettner ,

Please take a look at the CX related to discussion around the ticket https://github.com/camunda/team-connectors/issues/479.

Thank you.

File: connectors/aws/aws-base/src/main/java/io/camunda/connector/aws/model/impl/AwsBaseRequest.java [hunk 1]
```diff
@@ -7,20 +7,19 @@
 package io.camunda.connector.aws.model.impl;
 
 import jakarta.validation.Valid;
-import jakarta.validation.constraints.NotNull;
 import java.util.Objects;
 
 public class AwsBaseRequest {
 
-  @Valid @NotNull private AwsBaseAuthentication authentication;
+  @Valid private AwsAuthentication authentication;
```
[the remark this answers is not in the document]
Comment (sbuettner) [reply connectors/aws/aws-base/src/main/java/io/camunda/connector/aws/model/impl/AwsBaseRequest.java:14]: Or do you have to allow `null` due to the introduction of the type?

[the remark this answers is not in the document]
Comment (igpetrov) [reply connectors/aws/aws-base/src/main/java/io/camunda/connector/aws/model/impl/AwsBaseRequest.java:14]: Good catch. No, actually I lost it in refactoring. It should resolve to an AwsAuthentication anyway.

File: connectors/aws/aws-base/src/main/java/io/camunda/connector/aws/CredentialsProviderSupport.java [hunk 2]
```diff
@@ -9,14 +9,25 @@
...
-    AwsBaseAuthentication authentication = request.getAuthentication();
-    return new AWSStaticCredentialsProvider(
-        new BasicAWSCredentials(authentication.getAccessKey(), authentication.getSecretKey()));
+    AwsAuthentication authentication = request.getAuthentication();
+    if (authentication instanceof AwsStaticCredentialsAuthentication sca) {
+      LOGGER.debug("Using AwsStaticCredentialsAuthentication for AWS authentication");
+      return new AWSStaticCredentialsProvider(
+          new BasicAWSCredentials(sca.accessKey(), sca.secretKey()));
+    }
+    LOGGER.debug("Falling to DefaultAWSCredentialsProviderChain for AWS authentication");
```
Comment (sbuettner) [inline connectors/aws/aws-base/src/main/java/io/camunda/connector/aws/CredentialsProviderSupport.java:30]: Is that behaviour compatible with existing templates?

**Review by sbuettner — DISMISSED** [review]
LGTM
* minor comment regarding  validation

File: connectors/aws/aws-base/src/main/java/io/camunda/connector/aws/model/impl/AwsAuthentication.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+package io.camunda.connector.aws.model.impl;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+// Note on `defaultImpl`. This is left for backwards compatibility
+// with existing BPMN diagrams.
+@JsonTypeInfo(
+    use = JsonTypeInfo.Id.NAME,
+    property = "authenticationType",
```
[the remark this answers is not in the document]
Comment (igpetrov) [reply connectors/aws/aws-base/src/main/java/io/camunda/connector/aws/model/impl/AwsAuthentication.java:16]: Done

File: connectors/aws/aws-base/src/main/java/io/camunda/connector/aws/model/impl/AwsBaseRequest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (igpetrov) [reply connectors/aws/aws-base/src/main/java/io/camunda/connector/aws/model/impl/AwsBaseRequest.java:14]: Done in next rev

**Review by YanaSegal — APPROVED** [review]
From UX perspective LGTM. As Igor mentioned, this specific use case will be included in the documentation as part of a separate issue. Therefore, if users need clarification about the authentications they need to use, they can find their way around by referring to the documentation.

**Pull request merged**